packages/bindings-test/src/multitest.rs

```diff
@@ -540,4 +592,128 @@ mod tests {
         ];
         assert_eq!(state.assets, expected_assets);
     }
+
+    #[test]
+    fn perform_swap_with_route_exact_out() {
+        let pool1 = Pool::new(coin(6_000_000, "osmo"), coin(3_000_000, "atom"));
+        let pool2 = Pool::new(coin(2_000_000, "atom"), coin(1_000_000, "btc"));
+        let trader = Addr::unchecked("trader");
+
+        // set up pools
+        let mut app = OsmosisApp::new();
+        app.init_modules(|router, _, storage| {
+            router.custom.set_pool(storage, 1, &pool1).unwrap();
+            router.custom.set_pool(storage, 2, &pool2).unwrap();
+            router
+                .bank
+                .init_balance(storage, &trader, coins(5000, "osmo"))
+                .unwrap()
+        });
+
+        // now a proper swap
+        let msg = OsmosisMsg::Swap {
+            first: Swap {
+                pool_id: 1,
+                denom_in: "osmo".to_string(),
+                denom_out: "atom".to_string(),
+            },
+            route: vec![Step {
+                pool_id: 2,
+                denom_out: "btc".to_string(),
+            }],
+            amount: SwapAmountWithLimit::ExactOut {
+                output: Uint128::new(1000),
+                max_input: Uint128::new(5000),
+            },
+        };
+        let res = app.execute(trader.clone(), msg.into()).unwrap();
+
+        let Coin { amount, .. } = app.wrap().query_balance(&trader, "osmo").unwrap();
+        assert_eq!(amount, Uint128::new(5000 - 4033));
+        let Coin { amount, .. } = app.wrap().query_balance(&trader, "btc").unwrap();
+        assert_eq!(amount, Uint128::new(1000));
+
+        // check the response contains proper value
+        let input: EstimatePriceResponse = from_slice(res.data.unwrap().as_slice()).unwrap();
```

Comment:
This should be called `SwapResponse`. Changing it is kind of difficult now, as it'll impact osmosis.